cookiecutter/cli.py

```diff
@@ -67,4 +68,9 @@ def main(template, no_input, checkout, verbose):
             level=logging.INFO
         )
 
-    cookiecutter(template, checkout, no_input)
+    if extra_context:
+        # Convert tuple -- e.g.: (u'program_name=foobar',)
+        # to dict -- e.g.: {'program_name': 'foobar'}
+        extra_context = dict(s.split('=', 1) for s in extra_context)
```

Comment:
I just added a validation callback. Output looks like this:

```
$ cookiecutter --no-input https://github.com/msabramo/cookiecutter-supervisor.git program_name=foobar startsecs=66 foobar
Usage: cookiecutter [OPTIONS] TEMPLATE [EXTRA_CONTEXT]...

Error: Invalid value for "extra_context": EXTRA_CONTEXT should contain items of form key=value; 'foobar' doesn't match that form
```

How does that look?